tests/quickcheck.rs

```diff
@@ -0,0 +1,127 @@
+extern crate quickcheck;
+#[allow(unused_imports)]
+#[macro_use]
+extern crate quickcheck_macros;
+
+#[cfg(any(feature = "xxh32", feature = "xxh64", feature = "xxh3"))]
+mod tests {
+    use quickcheck::TestResult;
+    use std::hash::Hasher;
+    use std::num::{NonZeroU8, NonZeroUsize};
+    use xxhash_c_sys as sys;
+
+    // In practice 2048 bytes of data should cover all cases for the streaming hashers.
+    // So we use a limit 10 times that to cover more chunking variations.
+    const MAX_STREAM_SIZE: usize = 2048 * 10;
+
+    #[cfg(feature = "xxh3")]
+    #[quickcheck]
+    fn xxh3_chunked_matches_buffered(
+        chunk_size: NonZeroUsize,
+        xs: Vec<u8>,
+        times: NonZeroU8,
+        additional: u8,
+    ) -> TestResult {
+        // additional argument doubles down as the hasher seed
+        let seed = additional as u64;
+        // the vecs produced by quickcheck are perhaps a bit small by default.
+        // additional should add some noise to avoid only getting nice even lengths.
+        let target_size = (xs.len() * times.get() as usize + additional as usize) % MAX_STREAM_SIZE;
+        let xs = xs.into_iter().cycle().take(target_size).collect::<Vec<_>>();
+
+        // write all at once
+        let mut h0 = xxhash_rust::xxh3::Xxh3::with_seed(seed);
+        h0.write(&xs);
+        let h0 = h0.finish();
+
+        // write in chunks
+        let mut h1 = xxhash_rust::xxh3::Xxh3::with_seed(seed);
+        for chunk in xs.chunks(chunk_size.get()) {
+            h1.write(chunk);
+        }
+        let h1 = h1.finish();
+
+        let one_shot_result = xxhash_rust::xxh3::xxh3_64_with_seed(&xs, seed);
+
+        let sys_result = unsafe { sys::XXH3_64bits_withSeed(xs.as_ptr() as _, xs.len(), seed) };
+
+        // compare all, including to buffered and reference
+        let outcome = h0 == h1 && h0 == one_shot_result && h0 == sys_result;
```

Comment:
Is it possible to make comparison more precise?
I'm not familiar with quick check, but the way it is done there is no way to know what went wrong exactly without breaking this test apart

Reply:
Good point, I changed it to a standard assert_eq. On failures we'd get something like 
```
thread 'tests::xxh64_chunked_matches_buffered' panicked at '[quickcheck] TEST FAILED (runtime error). Arguments: (1, [], 1, 0)
Error: "assertion failed: `(left != right)`\n  left: `17241709254077376921`,\n right: `17241709254077376921`"
```